lib/galaxy/util/__init__.py

```diff
@@ -848,12 +848,20 @@ def unicodify( value, encoding=DEFAULT_ENCODING, error='replace', default=None )
     """
     Returns a unicode string or None
     """
-
-    if isinstance( value, text_type ):
-        return value
+    if value is None:
+        return None
+    if isinstance(value, bool):
+        return text_type(value)
+    if isinstance( value, string_types):
+        if isinstance( value, text_type ):
+            return value
+        else:
+            return text_type( value, encoding, error )
     try:
-        return text_type( str( value ), encoding, error )
-    except:
+        return text_type( value, encoding, error )
+    except Exception as e:
+        log.debug("value %s could not be coerced to unicode" % value)
+        log.debug(e)
         return default
```

Comment:
This implementation returns `default` if `value` is e.g. a list, which is wrong. I think the following implementation is more correct:
```
    if value is None:
        return None
    if not isinstance(value, string_types):
        try:
            value = str(value)
        except Exception:
            log.exception("value %s could not be coerced to unicode" % value)
            return default
    # At this point value is of type str, which in Python 2 needs to be converted to unicode
    if not isinstance(value, text_type):
        value = text_type(value, encoding, error)
    return value
```